src/components/InterruptionSettings.js

```diff
@@ -0,0 +1,334 @@
+    websitesCopy.websites.forEach((wc) => {
+      if (wc.name === name) {
+        wc.state = !wc.state;
+      }
+    });
+    setSharedState(websitesCopy);
+    props.updateWebsites(websitesCopy);
+    changingInterval();
+  };
+
+  const toggleCheckboxes = (name) => {
+    let websitesCopy = { ...sharedState };
+    websitesCopy.websites.forEach((wc) => {
+      if (wc.name !== name) {
+        wc.state = false;
+      } else {
+        wc.state = true;
+      }
+    });
+    setSharedState(websitesCopy);
+    props.updateWebsites(websitesCopy);
+  };
+
+  const validateInput = (name) => {
+    let websitesCopy = { ...sharedState };
+    websitesCopy.websites.forEach((wc) => {
+      if (wc.name === name) {
+        if (wc.interval < 5) {
+          wc.interval = 5;
+        } else if (wc.interval > 60) {
+          wc.interval = 60;
+        }
+      }
+    });
+    setSharedState(websitesCopy);
+    props.updateWebsites(websitesCopy);
+  };
+
+  const disableField = (name) => {
+    let websitesCopy = { ...sharedState };
+    websitesCopy.websites.forEach((wc) => {
+      if (wc.name === name) {
+        if (wc.isDisabled) {
+          wc.state = true;
+        } else {
+          wc.state = false;
+        }
+        wc.isDisabled = !wc.isDisabled;
+      }
+    });
+    setSharedState(websitesCopy);
+    props.updateWebsites(websitesCopy);
+    disableSlider();
+  };
+
+  const disableSlider = () => {
+    if (
+      sharedState.websites.every((w) => w.isDisabled) ||
+      sharedState.websites.length === 0
+    ) {
+      setAllDisabled(true);
+    } else {
+      setAllDisabled(false);
+    }
+  };
+
+  const usePrevious = (value) => {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
+    return ref.current;
+  };
+
+  const prevLoading = usePrevious(isLoading);
+
+  useEffect(() => {
+    if (!prevLoading && isLoading) {
+      disableSlider();
+      setIsLoading(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isLoading]);
+
+  const deleteWebsite = (name) => {
+    let websitesCopy = { ...sharedState };
+    let newWebsites = websitesCopy.websites.filter((wc) => wc.name !== name);
+    websitesCopy.websites = newWebsites;
+    setSharedState(websitesCopy);
+    props.updateWebsites(websitesCopy);
+    setIsLoading(true);
+  };
+
+  const addWebsite = () => {
+    if (
+      sharedState.websites.some(
+        (ws) => ws.name === nameInput || ws.URL === urlInput
+      )
+    ) {
+      alert("This name/URL has already been stored");
+    } else {
+      let newWebsite = {
+        name: nameInput,
+        URL: urlInput,
+        state: true,
+        interval: 15,
+        isDisabled: false,
+      };
+      let websitesCopy = { ...sharedState };
+      websitesCopy.websites.push(newWebsite);
+      setSharedState(websitesCopy);
+      props.updateWebsites(websitesCopy);
+    }
+    setNameInput("");
+    setURLInput("https://");
+  };
+
+  const options = {
```

Comment:
this is quite a big file... worth considering making it a bit smaller; the low hanging fruit is this definition that could be extracted to it's own file.